independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanProcessor.java

```diff
@@ -291,7 +292,29 @@ public Builder setName(String name) {
         }
 
         public Builder setIndex(IndexView index) {
-            this.index = index;
+            return setBeanArchiveIndex(index);
+        }
+
+        /**
+         * Set the bean archive index used to discover beans, interceptors, qualifiers, etc.
+         * 
+         * @param beanArchiveIndex
+         * @return self
+         */
+        public Builder setBeanArchiveIndex(IndexView beanArchiveIndex) {
+            this.beanArchiveIndex = beanArchiveIndex;
+            return this;
+        }
+
+        /**
+         * Set the (optional) additional index used to discover types during type-safe resolution. These types may not be part
+         * of the bean archive index.
+         * 
+         * @param additionalIndex
+         * @return self
+         */
+        public Builder setAdditionalIndex(IndexView additionalIndex) {
```

Comment:
I decided to use a more general name (`additionalIndex`) because it may be useful for other use cases as well...

Reply:
Perhaps `applicationIndex` would be a better name?

Reply:
Hmm, that's a better name, yes...